src/lib.rs

```diff
@@ -68,16 +61,31 @@ impl<AccountId, Moment> Shipment<AccountId, Moment> {
     }
 }
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub enum ShippingOperation {
+    Pickup,
+    Delivery,
+}
+
 #[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
 pub enum ShippingEventType {
+    ShipmentRegistration,
     ShipmentPickup,
     ShipmentDelivery,
```

Comment:
```suggestion
    ShipmentInTransit,
    ShipmentDelivered,
```

See above on reasoning